### PR TITLE
[receiver/signalfx]fix lint for signalfxreceiver

### PR DESCRIPTION
**Pull request by fatsheep9146**
Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

**Description:** <Describe what has changed.>
fix errcheck issue for signalfxreceiver

**Link to tracking Issue:** <Issue number if applicable>
#9749

### CODE REVIEW

File: receiver/signalfxreceiver/receiver.go [hunk 1]
```diff
@@ -215,7 +214,7 @@ func (r *sfxReceiver) writeResponse(ctx context.Context, resp http.ResponseWrite
 	}
 
 	resp.WriteHeader(http.StatusOK)
-	resp.Write(okRespBody)
+	_, _ = resp.Write(okRespBody)
```
Comment (TylerHelmuth) [inline receiver/signalfxreceiver/receiver.go:217]: Should we be checking that no error is returned or is it ok to ignore it?

File: receiver/signalfxreceiver/receiver_test.go [hunk 2]
```diff
@@ -178,8 +179,10 @@ func Test_signalfxeceiver_EndToEnd(t *testing.T) {
 		}
 		return false
 	}, 10*time.Second, 5*time.Millisecond, "failed to wait for the port to be open")
-	defer exp.Shutdown(context.Background())
-	require.NoError(t, exp.ConsumeMetrics(context.Background(), want))
+	defer func() {
+		require.NoError(t, exp.Shutdown(context.Background()))
+	}()
+	defer require.NoError(t, exp.ConsumeMetrics(context.Background(), want))
```
Comment (dmitryax) [inline receiver/signalfxreceiver/receiver_test.go:185]: Why defer added here? 
```suggestion
	require.NoError(t, exp.ConsumeMetrics(context.Background(), want))
```

File: receiver/signalfxreceiver/receiver.go (hunk 1 above)
Comment (dmitryax) [reply receiver/signalfxreceiver/receiver.go:217]: It's ok to ignore, there is no established way to report error in this function. At least we ignore it explicitly now

Comment (fatsheep9146) [reply receiver/signalfxreceiver/receiver.go]: I use failRequest function to handle this error, is that ok?

**Review by dmitryax — APPROVED** [review]

**Pull request merged**